### ISSUE TITLE
[typescript-operations] generates wrong types for query variables

### ISSUE THREAD

**Issue by lorado** [dotansimha/graphql-code-generator#5479]
**Describe the bug**
Since version 1.17.13 `@graphql-codegen/typescript-operations` generates for query with variable like
```graphql
query user($roles: [String!]) {
    #...
}
```
following: 
```ts
export type UserQueryVariables = Exact<{
  roles?: Maybe<Array<Scalars["String"]> | Scalars["String"]>;
}>;
```
instead of `Maybe<Array<Scalars["String"]>>` 
On Version 1.17.12 it works correctly. This issue occurs only on query variable types. If such type is defined in the schema, it is processed properly by `typescript` plugin. 

**To Reproduce**
Here is a codesandbox link with generated output with 1.17.14 in the file `types.ts` and 1.17.12 in the file `types_1.17.12.ts`
https://codesandbox.io/s/buggraphql-codegentypescript-operations-tspuu

1. My GraphQL schema:

```graphql
type Query {
    users(roles: [String!]): [User!]!
}

type User {
    id: ID!
    username: String!
    email: String!
    roles: [String!]
}
```

2. My GraphQL operations:

```graphql
query user($roles: [String!]) {
    users(roles: $roles) {
        id
        username
        email
    }
}

```

3. My `codegen.yml` config file:

```yml
schema: schema.graphql
documents: document.graphql
generates:
  types.ts:
    plugins:
      - typescript
      - typescript-operations

```

**Expected behavior**
`query user($roles: [String!]) {...}`  should be mapped as following type:

```ts
export type UserQueryVariables = Exact<{
  roles?: Maybe<Array<Scalars["String"]>>;
}>;
```

**Environment:**

- OS: MacOS
- `@graphql-codegen/typescript-operations`: 1.17.14
- NodeJS: 14

**Additional context**

**Comment by campsafari** [comment]
Same here.

**Comment by lorado** [comment]
Ok... if you guys facing this "problem" then you probably missed something, like me... 

This behavior is intended and is documented in the GraphQL specification:
- in latest spec from June 2018 - https://spec.graphql.org/June2018/#sec-Type-System.List
- in draft spec -  https://spec.graphql.org/draft/#sec-Type-System.List.Input-Coercion

`typescript-operations` plugin was missing this feature, and this was mentioned in #4888 (definitely take a look into this!), fixed in #5115. 

There is a workaround type to use, that can be found here: https://github.com/dotansimha/graphql-code-generator/issues/5352#issuecomment-775851219

I am closing this issue, as it turned out, this is not a bug, this is a spec definition.
